Clean Swift/Unit Tests (XCTest).xctemplate/___FILEBASENAME___InteractorTests.swift

```diff
@@ -101,53 +127,55 @@ class ___VARIABLE_sceneName___InteractorTests: XCTestCase {
     }
-    
-    func testPerform___VARIABLE_sceneName___ShouldAskPresenterToFormat() {
+
+    func testPerform___VARIABLE_sceneName___ShouldNotAskWorkerToPerform___VARIABLE_sceneName___IfThereAreErrors() {
         // given
-        let request = ___VARIABLE_sceneName___Models.Perform___VARIABLE_sceneName___.Request()
-        let spy = ___VARIABLE_sceneName___PresentationLogicSpy()
-        sut.presenter = spy
-        
+        let spy = ___VARIABLE_sceneName___WorkerSpy()
+        sut.worker = spy
+        let request = ___VARIABLE_sceneName___Models.Perform___VARIABLE_sceneName___.Request(exampleVariable: nil)
+
         // when
         sut.perform___VARIABLE_sceneName___(with: request)
-        
+
         // then
-        XCTAssertTrue(spy.presentPerform___VARIABLE_sceneName___Called, "perform___VARIABLE_sceneName___(with:) should ask the presenter to format the result")
+        XCTAssertFalse(spy.perform___VARIABLE_sceneName___Called, "perform___VARIABLE_sceneName___(with:) should not ask the worker to perform ___VARIABLE_sceneName___")
     }
-    
-    func testPerform___VARIABLE_sceneName___ShouldNotAskWorkerToPerform___VARIABLE_sceneName___IfThereIsAnError() {
+
+    func testPerform___VARIABLE_sceneName___ShouldAskWorkerToPerform___VARIABLE_sceneName___IfThereAreNoErrors() {
         // given
-        let request = ___VARIABLE_sceneName___Models.Perform___VARIABLE_sceneName___.Request(exampleVariable: nil)
         let spy = ___VARIABLE_sceneName___WorkerSpy()
         sut.worker = spy
-        
+        let request = ___VARIABLE_sceneName___Models.Perform___VARIABLE_sceneName___.Request(exampleVariable: "Example string.")
+
         // when
         sut.perform___VARIABLE_sceneName___(with: request)
-        
+
         // then
-        XCTAssertFalse(spy.perform___VARIABLE_sceneName___Called, "perform___VARIABLE_sceneName___(with:) should not ask worker to perform ___VARIABLE_sceneName___ if there is an error")
+        XCTAssertTrue(spy.perform___VARIABLE_sceneName___Called, "perform___VARIABLE_sceneName___(with:) should ask the worker to perform ___VARIABLE_sceneName___")
     }
-    
-    func testPerform___VARIABLE_sceneName___ShouldAskWorkerToPerform___VARIABLE_sceneName___IfThereIsNoError() {
+
+    func testPerform___VARIABLE_sceneName___ShouldAskPresenterToFormat() {
         // given
-        let request = ___VARIABLE_sceneName___Models.Perform___VARIABLE_sceneName___.Request(exampleVariable: "Example string.")
-        let spy = ___VARIABLE_sceneName___WorkerSpy()
-        sut.worker = spy
-        
+        let spy = ___VARIABLE_sceneName___PresentationLogicSpy()
+        sut.presenter = spy
+        let request = ___VARIABLE_sceneName___Models.Perform___VARIABLE_sceneName___.Request()
+
         // when
         sut.perform___VARIABLE_sceneName___(with: request)
-        
+        expect.fulfill()
+        waitForExpectations(timeout: 1)
```

Comment:
Any reason we should _always_ have waiting for expectations?

Reply:
@zaimramlan this is the equivalent to `.toEventually`, I just direct translate from Spec